apps/platform-app/src/app/[locale]/(profile)/_actions/queries/findProfileByLinkedinUsername.ts

```diff
@@ -0,0 +1,18 @@
+'use server'
+
+import { createProfileModel } from '@/app/[locale]/(profile)/_models/profile.model'
+import { getProfileByLinkedinUsername } from '@/backend/profile/profile.service'
+import { withSentry } from '@/utils/errHandling'
+
+export const findProfileByLinkedinUsername = withSentry(
```

Comment:
**nit:** I don't see point of separating these on `findByProviderNameUsername` as in terms of how it works in our platform username is username, so I would go with `findUserByUsername`, and then do:

```ts

prisma.users.findFirst({
  OR: [{githubDetails: { username }}, {linkedinDetails: { username }}]
})

```

Reply:
> Profile URL Conflicts: In cases where users across providers share the same username (e.g., one as a GitHub username and another as a LinkedIn username), profile URL conflicts may arise.

i think we should rething the whole idea of creting profile url's as yhis approach may get us some trouble later

Reply:
yea - that's why we should build our own slug IMO - so adding unique `username` to User model would be safest approach. then we could have single usernames because we could use github usernames to prefill user usernames and unify everyhing